blog/2025-03-25-sentry-openfeature-integration.md

```diff
@@ -0,0 +1,102 @@
+---
+slug: "sentry-openfeature-integration"
+title: "Sentry OpenFeature Integration"
+date: 2025-04-08
+authors: [jonathannorris]
+description: "Announcing Sentry's OpenFeature integration for enhanced feature flag observability"
+tags: [sentry, openfeature, "feature flags", observability]
+draft: false
+---
+
+We're excited to announce that Sentry has released OpenFeature Hooks for [JavaScript](https://docs.sentry.io/platforms/javascript/configuration/integrations/openfeature/) and [Python](https://docs.sentry.io/platforms/python/integrations/openfeature/), enabling developers to track feature flag evaluations directly in their error monitoring and performance tracking.
+
+## What is the Sentry OpenFeature Integration?
+
+The Sentry OpenFeature Hook automatically tracks feature flag evaluations in your application.
+When an error occurs, Sentry will include the state of all evaluated feature flags in the error report, providing crucial context for debugging and troubleshooting.
+
+This integration is currently in beta and only supports boolean flag evaluations.
+
+## JavaScript Integration
+
+To use the Sentry OpenFeature Hook in your JavaScript application:
+
+```typescript
+import * as Sentry from "@sentry/browser";
+import { OpenFeature } from "@openfeature/web-sdk";
+
+Sentry.init({
+  dsn: "https://examplePublicKey@o0.ingest.sentry.io/0",
+  integrations: [Sentry.openFeatureIntegration()],
+});
+
+OpenFeature.setProvider(new MyProviderOfChoice());
+OpenFeature.addHooks(new Sentry.OpenFeatureIntegrationHook());
+
+const client = OpenFeature.getClient();
+const result = client.getBooleanValue("test-flag", false); // evaluate with a default value
+Sentry.captureException(new Error("Something went wrong!"));
+```
+
+> Requirements: Sentry SDK version 8.43.0 or higher
+
+## Python Integration
+
+For Python applications, you can use the Sentry OpenFeature Hook as follows:
+
+```python
+import sentry_sdk
+from sentry_sdk.integrations.openfeature import OpenFeatureIntegration
+from openfeature import api
+
+// Initialize Sentry with the OpenFeature integration
+sentry_sdk.init(
+    dsn="https://examplePublicKey@o0.ingest.sentry.io/0",
+    integrations=[
+        OpenFeatureIntegration(),
+    ],
+)
+
+// Get your OpenFeature client
+client = api.get_client()
+
+// Evaluate a feature flag
+client.get_boolean_value("hello", default_value=False)
+
+// If an error occurs, Sentry will include the feature flag state
+sentry_sdk.capture_exception(Exception("Something went wrong!"))
+```
+
+> Requirements: sentry-sdk >= 2.19.2, openfeature-sdk >= 0.7.1, Python >= 3.8
+
+## Benefits of Feature Flag Observability
```

Comment:
Maybe it would be helpful to give an example (screenshot) how it looks like in the Sentry portal?